modules/storage-page-memory/src/main/java/org/apache/ignite/internal/storage/pagememory/mv/PersistentPageMemoryMvPartitionStorage.java

```diff
@@ -351,32 +351,15 @@ protected List<AutoCloseable> getResourcesToClose() {
     /**
      * Syncs and saves meta-information on checkpoint.
      *
-     * @param executor Executor for asynchronous data synchronization.
-     * @throws IgniteInternalCheckedException If failed.
+     * @param executor Executor for asynchronous data synchronization, {@code null} if absent.
      */
-    private void syncMetadataOnCheckpoint(@Nullable Executor executor) throws IgniteInternalCheckedException {
+    private void syncMetadataOnCheckpoint(@Nullable Executor executor) {
         RenewablePartitionStorageState localState = renewableState;
 
         if (executor == null) {
-            localState.rowVersionFreeList().saveMetadata();
-
-            localState.indexFreeList().saveMetadata();
+            saveFreeListsMetadata(localState);
         } else {
-            executor.execute(() -> {
-                try {
-                    localState.rowVersionFreeList().saveMetadata();
-                } catch (IgniteInternalCheckedException e) {
-                    throw new StorageException("Failed to save RowVersionFreeList metadata", e);
-                }
-            });
-
-            executor.execute(() -> {
-                try {
-                    localState.indexFreeList().saveMetadata();
-                } catch (IgniteInternalCheckedException e) {
-                    throw new StorageException("Failed to save IndexColumnsFreeList metadata", e);
-                }
-            });
+            executor.execute(() -> saveFreeListsMetadata(localState));
```

Comment:
There were two parallel calls before, not there's only one. Why did you change that?

Reply:
Fix it